paddle/fluid/framework/ir/fc_cudnn_gru_fuse_pass.cc

```diff
@@ -0,0 +1,260 @@
+    auto* scope = graph->Get<Scope*>(kParamScopeAttr);
+
+    const std::string CacheNode = patterns::UniqueKey("Cache");
+
+    VarDesc cudnn_gru_w_desc(patterns::PDNodeName(name_scope, "W"));
+    cudnn_gru_w_desc.SetPersistable(true);
+    auto* cudnn_gru_w_node = graph->CreateVarNode(&cudnn_gru_w_desc);
+    auto* w_tensor =
+        scope->Var(cudnn_gru_w_node->Name())->GetMutable<LoDTensor>();
+
+    // create a inith node wizh size=0
+    VarDesc cudnn_gru_inith_desc(patterns::PDNodeName(name_scope, "InitH"));
+    cudnn_gru_inith_desc.SetPersistable(true);
+    auto* cudnn_gru_inith_node = graph->CreateVarNode(&cudnn_gru_inith_desc);
+    auto* h_tensor =
+        scope->Var(cudnn_gru_inith_node->Name())->GetMutable<LoDTensor>();
+    std::vector<int64_t> ha = {0, 0};
+    h_tensor->Resize(framework::make_ddim(ha));
+    std::fill_n(h_tensor->mutable_data<float>(platform::CPUPlace()),
+                h_tensor->numel(), 0.0f);
+
+    scope->Var(CacheNode);
+    op_desc.SetInput("Cache", {CacheNode});
+
+    op_desc.SetInput("W", {cudnn_gru_w_node->Name()});
+
+    op_desc.SetInput("InitH", {cudnn_gru_inith_node->Name()});
+
+    auto* weight_x_var = scope->FindVar(weight_x->Name());
+    auto* weight_h_var = scope->FindVar(weight_h->Name());
+
+    auto* weight_x_tensor = weight_x_var->GetMutable<framework::LoDTensor>();
+    auto* weight_h_tensor = weight_h_var->GetMutable<framework::LoDTensor>();
+    auto* gru_bias_var = scope->FindVar(bias->Name());
+    auto* gru_bias_tensor = gru_bias_var->GetMutable<framework::LoDTensor>();
+
+    int input_size = weight_x_tensor->dims()[0];
+    int hidden_size = weight_h_tensor->dims()[0];
+
+    weight_x_tensor->mutable_data<float>(platform::CPUPlace());
+    weight_h_tensor->mutable_data<float>(platform::CPUPlace());
+    gru_bias_tensor->mutable_data<float>(platform::CPUPlace());
+
+    op_desc.SetAttr("input_size", input_size);
+    op_desc.SetAttr("hidden_size", hidden_size);
+    op_desc.SetAttr("max_len", 64);
+
+    std::vector<int64_t> wdim = {weight_x_tensor->dims()[0] * 3 +
+                                     weight_h_tensor->dims()[1] +
+                                     gru_bias_tensor->dims()[0] * 6,
+                                 weight_h_tensor->dims()[0]};
+
+    framework::DDim dim = framework::make_ddim(wdim);
+    w_tensor->Resize(dim);
+
+    w_tensor->mutable_data<float>(platform::CPUPlace());
+
+    // copy wx data
+    int offset = weight_x_tensor->dims()[0] * weight_h_tensor->dims()[0];
+
+    // copy reset gate furst
+    float* w_data = w_tensor->data<float>();
+
+    memcpy(w_data, weight_x_tensor->data<float>() + offset,
+           offset * sizeof(float));
+    w_data += offset;
+
+    // copy update gate data
+    memcpy(w_data, weight_x_tensor->data<float>(), offset * sizeof(float));
+    w_data += offset;
+
+    memcpy(w_data, weight_x_tensor->data<float>() + 2 * offset,
+           offset * sizeof(float));
+    w_data += offset;
+
+    // copy wh data
```

Comment:
This long function mixes too many operations:

- Prepare weight
- construct the OpDesc

Each one is very long, please split to several shorter functions.